### DISCUSSION TITLE
Outputting formatted time

### DISCUSSION THREAD

**Discussion by rwl7532** [jkotlinski/durexforth#535] [Q&A] [answered]
If about to put 6 24 34 ( hrs min sec ) on the stack. 
These numbers are pulled from TOD registers, converting bcd to numbers.
I want the output to the screen to be 06:24:34
Near as I can tell after the stack is filled correctly, the output involves <# # # 58 hold # # 58 hold # # #> type ( or something similar )
I keep failing. Help? Is s>d involved?

**Comment by Whammo** [comment]
You seem to be on the right track.
For more info about number formatting, read Starting Forth.

**Reply by rwl7532** [reply]
Can what I want be done interactively or must I write some words to accomplish it? Every interactive I'm trying is failing. Failing really hard. Starting FORTH permits 6. 24. 34. durexforth does not.

**Reply by Whammo** [reply]
I do know that number formatting works on doubles. Beyond that, I have not explored the possibilities.

**Reply by Whammo** [reply]
Also  I believe it returns a string  not numbers.

**Reply by jkotlinski** [reply]
Input conversion of double-length numbers (e.g. `1234.`) is not supported by durexForth. Instead, type either `1234 0` or `1234 s>d` to put the double-length number `1234 0` on the stack. I hope that helps!

**Reply by rwl7532** [reply]
Will try and report.

**Comment by Whammo** [comment]
Here is my ``ptod`` that finally does the colons. Without using FORMAT:
 
 ```forth
: rtod $dd0b dup c@ dup $7f and >r
         $80 and swap
         1- dup c@ >r
         1- dup c@ >r
         1-     c@ >r
         r> r> r> r> ;

: ptod  hex rtod          \ prints TOD
3 0 do . $9d emit ':' emit loop
. decimal $80 and if   ." pm"
                else ." am"
                then cr ;

**Reply by Whammo** [reply]
However a leading zero for two places. would be preferable for minutes and seconds.

**Comment by rwl7532** [comment]
:  hrmin <# # # 58 hold # # #> type ;
10 0 54 0 [Enter] 
hrmin
00:54ok
hrmin
00:10ok

Something in my understanding is wrong.

**Reply by jkotlinski** [reply]
The problem is, `hrmin` does not move on from one number to the next, until it reaches `#>`. This would work better:

    : 00. s>d <# # # #> type ;
    : hrmin swap 00. ':' emit 00. ;
    10 54 hrmin
    10:54ok

**Reply by rwl7532** [reply]
I'm going with this!! I'm adding cr so ok appears on a new line.
Many thanks.

**Reply by rwl7532** [reply]
Why doesn't "Mark as answer" appear?

**Reply by Whammo** [reply]
Just one of the peculiarities of the platform. Comments are not answers.

**Reply by Whammo** [reply]
Also, when you have sorted it out, please share.

**Reply by rwl7532** [reply]
Willing to share but new to Commodore and just (today) got VICE working on my Windows 2007 laptop. VICE install on my Ubuntu fails to allow me to write to durexforth files. Just found viceutil but nothing about it when searching discussions.  
Any code in durexforth I can only share as screenshots. I'll be launching a discussion questioning what's up with how to share code etc.

**Comment by Whammo** [comment]
How I ended up.

```forth
: ptod  hex rtod          \ prints TOD
3 0 do dup 10 u< if '0' emit then
. $9d emit ':' emit loop
. decimal $80 and if   ." pm"
                else ." am"
                then cr ;
```

**Reply by Whammo** [reply]
A nice trick would be that if the space after 'dot' were a variable one could set to any character. 
 Or perhaps a string.

**Comment by rwl7532** [comment, marked as answer]
Answered! See above.

**Reply by ekipan** [reply]
Would be a bit easier to find if you marked [the other answer](https://github.com/jkotlinski/durexforth/discussions/535#discussioncomment-5458349) instead. Not a big deal though. I don't even know if you can change marked answers? Also new to this stuff.

**Reply by rwl7532** [reply]
"New to this stuff" fits me too!
